libs/components/modals/src/lib/modules/modal/modal.component.ts

```diff
@@ -35,69 +36,52 @@ let skyModalUniqueIdentifier = 0;
   selector: 'sky-modal',
   templateUrl: './modal.component.html',
   styleUrls: ['./modal.component.scss'],
-  animations: [skyAnimationModalState],
   providers: [SkyModalComponentAdapterService, SkyDockService],
 })
 export class SkyModalComponent implements AfterViewInit, OnDestroy {
   @HostBinding('class')
-  public get wrapperClass(): string {
-    return this.config.wrapperClass;
-  }
+  public wrapperClass: string | undefined;
 
+  // Ignoring coverage as we only use the setter internally and do not export the class externally for users to be able to use the getter.
+  // istanbul ignore next
   /**
    * @internal
    */
   @Input()
   public get ariaRole() {
-    return this.config.ariaRole || 'dialog';
+    return this.#_ariaRole;
   }
-  public set ariaRole(value: string) {
-    this.config.ariaRole = value;
+
+  public set ariaRole(value: string | undefined) {
+    this.#_ariaRole = value;
+    this.ariaRoleOrDefault = value || ARIA_ROLE_DEFAULT;
   }
 
+  public ariaRoleOrDefault = ARIA_ROLE_DEFAULT;
+
   /**
    * @internal
    */
   @Input()
-  public set tiledBody(value: boolean) {
-    this.config.tiledBody = value;
-  }
+  public tiledBody: boolean | undefined;
 
   public get modalZIndex() {
-    return this.hostService.getModalZIndex();
+    return this.#hostService.getModalZIndex();
```

Comment:
This seems wrong to me; can the z-index change during the lifecycle of the component, or is it constant once the modal is displayed? If it's the latter, we should get it once and assign it to a public property so we're not executing a function in a getter. If it's the former, we should use an Observable in the modal host service when it changes instead of relying on a change detection cycle to pick up the new value.

Reply:
I have reworked the modal host service logic to utilize a `ReplaySubject` and then have the modal component listen to updates from this. This allows us to listen to changes but also update the zIndex of each host correctly if a lower modal closes for some reason while another was on top of it (not a common scenario - but one I wanted to make sure was accounted for). I also added a unit test for this scenario.

Reply:
My first thought was just to reduce the complexity and get the z-index when the modal closes, but the current behavior does account for a lower modal being removed, so it's good that you preserved that behavior. I think you want a `BehaviorSubject` rather than a `ReplaySubject`, as `ReplaySubject` has an infinite buffer and will replay all previous values when a new subscriber is added, whereas `BehaviorSubject` only stores and emits the last value for new subscribers. You should also complete/close the subject when the modal host is destroyed so it doesn't hold onto old values indefinitely.

Reply:
We chatted in standup and decided to change this just to assign the zIndex when created based on the max of the already opened modals. I have updated things to do this.